Include/internal/pycore_critical_section.h

```diff
@@ -108,6 +109,25 @@ extern "C" {
         _PyCriticalSection2_End(&_cs2);                                 \
     }
 
+// Specialized version of critical section locking called to safely use
+// PySequence_Fast APIs under nogil
+// For performance, the argument *to* PySequence_Fast is provided to the
+// macro, not the *result* of PySequence_Fast (which would require an extra
+// test to determine if the lock must be held)
```

Comment:
Suggested minor edits:

```suggestion
// Specialized version of critical section locking to safely use
// PySequence_Fast APIs without the GIL. For performance, the argument *to*
// PySequence_Fast() is provided to the macro, not the *result* of
// PySequence_Fast(), which would require an extra test to determine if the
// lock must be acquired.
```